functions/src/modules/SQR.functions.ts

```diff
@@ -125,111 +125,130 @@ export const sendEmailOnNewAllotment = functions.database
-    let coordinator = functions.config().sqr.coordinator;
-    let templateId = functions.config().sqr.allotment.templateid;
-    // 3. Notify the coordinator
-    // 3.1 Get the submitted audio file data
+let prepareSubmissionParams = (original, audioFileStatus) => {
+  if (original.not_preferred_language) audioFileStatus = 'Spare';
+  else if (original.unable_to_play_or_download)
+    audioFileStatus = 'Audio Problem';
+
+  // 1. Add the webform data to a SQR submissions DB path
+  let submission = {
+    fileName: original.audio_file_name,
+    cancellation: {
+      notPreferredLanguage: audioFileStatus === 'Spare',
+      audioProblem: audioFileStatus === 'Audio Problem',
+    },
+    soundQualityRating: original.sound_quality_rating,
+    unwantedParts: original.unwanted_parts,
+    soundIssues: original.sound_issues,
+    duration: {
+      beginning: original.beginning,
+      ending: original.ending,
+    },
+    comments: original.comments,
+    token: original.token,
+    created: original.created,
+    //  timestamp of the submission creation,
+    // can differ from COMPLETED in case of saving a DRAFT and completing later.
+    completed: original.completed, // timestamp of the submission completion.
+    changed: original.changed, //timestamp of the submission update.
+    devotee: {
+      name: original.name,
+      emailAddress: original.email_address,
+    },
+  };
+
+  return submission;
+};
 
-    //  EXTRACTING the list name first from the file_name
-    let list = helpers.extractListFromFilename(original.audio_file_name);
+export const restructureExternalSubmission = functions.database
+.ref('/webforms/sqr/{submission_id}')
+.onCreate(async (snapshot, context) => {
+  const original = snapshot.val();
 
-    let fileSnapshot = await db
-      .ref(`/sqr/files/${list}/${original.audio_file_name}`)
-      .once('value');
+  let audioFileStatus = 'WIP';
+  let submission = prepareSubmissionParams(original, audioFileStatus);
 
-    if (fileSnapshot.exists()) {
-      let fileData = fileSnapshot.val();
+  db.ref(`/sqr/submissions/${original.serial}`).update(submission);
+  return 1;
+});
 
-      /////////////////////////////////////////////////////////////
-      //
-      // 3.2 Get the devotee's Allotments in ('given' || 'WIP') state
-      // TO BE ADDED LATER
-      // Currently passing an empty array
-      //
-      /////////////////////////////////////////////////////////////
+export const processSubmission = functions.database
+.ref('/sqr/submissions/{submission_id}')
+.onCreate(async (snapshot, context) => {
+  const original = snapshot.val();
 
-      // 3.3 checking if the First Submission or not
-      let submissionSnapshot = await db
-        .ref(`/sqr/submissions`)
-        .orderByChild('devotee/emailAddress')
-        .equalTo(original.email_address)
-        .once('value');
+  let audioFileStatus = 'WIP';
+  let submission = prepareSubmissionParams(original, audioFileStatus);
```

Comment:
In this function you should not prepare it, as the structure of the data at `/sqr/submissions` path is already normalized. Keeping this in mind, I don't think we need to extract `prepareSubmissionParams` as a separate function, you can do it directly inside `restructureExternalSubmission`.